pymatgen/core/surface.py

```diff
@@ -762,34 +745,19 @@ def center_slab(slab: Slab) -> Slab:
     Returns:
         Slab: The centered Slab.
     """
-    # Get all site indices
-    all_indices = list(range(len(slab)))
-
-    # Get a reasonable cutoff radius to sample neighbors
-    bond_dists = sorted(nn[1] for nn in slab.get_neighbors(slab[0], 10) if nn[1] > 0)
-    # TODO (@DanielYang59): magic number for cutoff radius (would 3 be too large?)
-    cutoff_radius = bond_dists[0] * 3
-
-    # TODO (@DanielYang59): do we need the following complex method?
-    # Why don't we just calculate the center of the Slab and move it to z=0.5?
-    # Before moving we need to ensure there is only one Slab layer though
-
-    # If structure is case 2, shift all the sites
-    # to the other side until it is case 1
-    for site in slab:  # DEBUG (@DanielYang59): Slab position changes during loop?
-        # DEBUG (@DanielYang59): sites below z=0 is not considered (only check coord > c)
-        if any(nn[1] >= slab.lattice.c for nn in slab.get_neighbors(site, cutoff_radius)):
-            # TODO (@DanielYang59): the magic offset "0.05" seems unnecessary,
-            # as the Slab would be centered later anyway
-            shift = 1 - site.frac_coords[2] + 0.05
-            slab.translate_sites(all_indices, [0, 0, shift])
```

Comment:
I think this function is overly complicated, we just need to:
1. Check the slab region to make sure there is only one continuous slab layer (with `get_slab_regions` function).
2. If the `slab` has multiple slab regions, it would be quite tricky as we're not sure if the slab region is broken by periodic boundary (a good `get_slab_regions` should handle this case though) or really being non-continuous.
3. Move the center of the slab to z=0.5 (fractional coordinate).

Reply:
feel free to refactor! not sure how good the test coverage is. if poor, it's often good to write additional tests first before refactoring to make sure no previous behavior was changed. or if changes are necessary, that we document all of them in the changelog

Reply:
Fold this for now, would address in a future PR (to avoid too many breaking changes in a single PR.)